ui/server.py

```diff
@@ -61,8 +67,69 @@ def _get_rules():
     return rulesdict, cat
 
 
+# Store JWT's value for every connected user
+registered_tokens = []
+
+
+@app.before_request
+def before_request():
+    """
+        Treat all incoming requests before-hand. 
+        If the user is not yet logged in, he/she will be redirected towards the login page.
+    """
+    if(HTTPS):
+        token = request.cookies.get('AUTH')
+        if(token not in registered_tokens):
+            if(request.endpoint != 'login' and '/res/' not in request.path):
```

Comment:
why is `/res/` needed in the request path?

Reply:
Thank you for the other comments.
Testing whether  `/res/` is part of the request.path or not is important here; there reason behind this is that the @app.before_request decorator will block access to any route besides /login; hence blocking access to these paths:

- https://0.0.0.0:5000/res/favicon/*
- https://0.0.0.0:5000/res/css/fonts.css
- https://0.0.0.0:5000/res/css/rules.css
- https://0.0.0.0:5000/res/*

Resulting in a pale page that has no CSS nor JS. That said, we make an exception to any path containing `/res/` .